compaction/compaction_manager.cc

```diff
@@ -1098,16 +1109,18 @@ class compaction_manager::offstrategy_compaction_task : public compaction_manage
                 co_await run_offstrategy_compaction(_compaction_data);
                 finish_compaction();
                 cmlog.info("Done with off-strategy compaction for {}.{}", t.schema()->ks_name(), t.schema()->cf_name());
-                co_return;
+                co_return std::nullopt;
             } catch (...) {
                 ex = std::current_exception();
             }
 
             finish_compaction(state::failed);
             if ((co_await maybe_retry(std::move(ex))) == stop_iteration::yes) {
-                co_return;
+                co_return std::nullopt;
             }
         }
+
+        co_return std::nullopt;
```

Comment:
I very much dislike this return value added for a single concrete task's sake. Not sure what would be the better alternative though.